src/sentry/api/endpoints/organization_invite_request_details.py

```diff
@@ -1,14 +1,202 @@
+
+        if (
+            organization.get_option("sentry:join_requests") is False
+            and member.invite_status == InviteStatus.REQUESTED_TO_JOIN.value
+        ):
+            raise serializers.ValidationError(ERR_JOIN_REQUESTS_DISABLED)
+
+        return approve
+
+
+class InviteRequestPermissions(OrganizationPermission):
+    scope_map = {
+        "GET": ["member:read", "member:write", "member:admin"],
+        "PUT": ["member:read", "member:write", "member:admin"],
+        "DELETE": ["member:read", "member:admin"],
+    }
 
 
 class OrganizationInviteRequestDetailsEndpoint(OrganizationEndpoint):
-    def get(self, request, organization):
-        # TODO(epurkhiser): Add listing of invite requests
-        pass
+    permission_classes = (InviteRequestPermissions,)
+
+    def _get_member(self, organization, member_id):
+        try:
+            return OrganizationMember.objects.get(
+                Q(invite_status=InviteStatus.REQUESTED_TO_BE_INVITED.value)
+                | Q(invite_status=InviteStatus.REQUESTED_TO_JOIN.value),
+                organization=organization,
+                user__isnull=True,
+                id=member_id,
+            )
+        except ValueError:
+            raise OrganizationMember.DoesNotExist()
+
+    def get(self, request, organization, member_id):
+        try:
+            member = self._get_member(organization, member_id)
+        except OrganizationMember.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        return Response(
+            serialize(member, serializer=OrganizationMemberWithTeamsSerializer()),
+            status=status.HTTP_200_OK,
+        )
+
+    def put(self, request, organization, member_id):
+        """
+        Update an invite request to Organization
+        ````````````````````````````````````````
+
+        Update and/or approve an invite request to an organization.
+
+        :pparam string organization_slug: the slug of the organization the member will belong to
+        :param string member_id: the member ID
+        :param boolean approve: allows the member to be invited
+        :param string role: the suggested role of the new member
+        :param array teams: the suggested slugs of the teams the member should belong to.
+
+        :auth: required
+        """
+
+        try:
+            member = self._get_member(organization, member_id)
+        except OrganizationMember.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        if (
+            member.inviter != request.user
+            and not request.access.has_scope("member:admin")
+            and not request.access.has_scope("member:write")
+        ):
+            return Response(
+                {"detail": ERR_INSUFFICIENT_ROLE_UPDATE}, status=status.HTTP_403_FORBIDDEN
+            )
+
+        serializer = OrganizationMemberSerializer(
+            data=request.data,
+            context={"organization": organization, "allowed_roles": roles.get_all()},
```

Comment:
allowed roles should probably be the users allowed roles, so this should be closer to what the member invite endpoint.

Reply:
updated. only added this check to the approve serializer, so members can still update invite requests regardless of the role.